cpu/nrf52/periph/i2c.c

```diff
@@ -99,20 +99,39 @@ void i2c_init(i2c_t dev)
     /* disable device during initialization, will be enabled when acquire is
      * called */
     bus(dev)->ENABLE = TWIM_ENABLE_ENABLE_Disabled;
-    /* configure pins */
-    gpio_init(i2c_config[dev].scl, GPIO_IN_OD_PU);
-    gpio_init(i2c_config[dev].sda, GPIO_IN_OD_PU);
-    bus(dev)->PSEL.SCL = i2c_config[dev].scl;
-    bus(dev)->PSEL.SDA = i2c_config[dev].sda;
+
     /* configure dev clock speed */
     bus(dev)->FREQUENCY = i2c_config[dev].speed;
 
     spi_twi_irq_register_i2c(bus(dev), i2c_isr_handler, (void *)dev);
 
-    /* re-enable the device. We expect that the device was being acquired before
+    /* configure pins */
+    /* also re-enables the device internally. We expect that the device was being acquired before
      * the i2c_init_master() function is called, so it should be enabled when
      * exiting this function. */
+    i2c_init_pins(dev);
+}
+
+void i2c_init_pins(i2c_t dev)
+{
+    assert(dev < I2C_NUMOF);
+
+    gpio_init(i2c_config[dev].scl, GPIO_IN_OD_PU);
+    gpio_init(i2c_config[dev].sda, GPIO_IN_OD_PU);
+    bus(dev)->PSEL.SCL = i2c_config[dev].scl;
+    bus(dev)->PSEL.SDA = i2c_config[dev].sda;
+
     bus(dev)->ENABLE = TWIM_ENABLE_ENABLE_Enabled;
+
+    mutex_unlock(&locks[dev]);
+}
+
+void i2c_deinit_pins(i2c_t dev)
+{
+    assert(dev < I2C_NUMOF);
+
+    mutex_lock(&locks[dev]);
+    bus(dev)->ENABLE = TWIM_ENABLE_ENABLE_Disabled;
```

Comment:
According to the commit that introduced these new functions to the I2C API (21958ff531e2619c3c3f037bb48f317bcce7c93b by @benpicco) they should be available only when the `periph_i2c_reconfigure` module is used. If that is the case they shouldn't be called from `i2c_init` and should be guarded.

This is how it's done for the other implementer of this module:
https://github.com/RIOT-OS/RIOT/blob/72621bcfcf20805f769150fc1d35cb14fd0929b3/cpu/sam0_common/periph/i2c.c#L179-L202

Reply:
I think it doesn't hurt to always have them available, if they are not used the linker will throw them out.
If it reduces code duplication, I think it's preferable this way.

Reply:
Ok, that's fine. I just wanted to be consistent with whatever migration was in place for this API extension.

Reply:
I mean we could also have a `static void _init_pins(i2c_t dev)` function that is called from both functions if that makes it cleaner.

Reply:
Yeah, I think that would be more consistent.

Reply:
Is this a change request or can we merge this feature as is?

Reply:
I'd like to be consistent and provide the feature only when the module is present. So I'd say let's adapt it before merging

Reply:
Actually the module is set in Makefile.features of nrf52. Doesn't that mean that it's always present?

Reply:
No, `FEATURES_PROVIDED += periph_i2c_reconfigure` means that the feature can be required (e.g. `FEATURES_REQUIRED += periph_i2c_reconfigure`), only then the module is pulled.

Reply:
Alright, I'll fix it now